Extensions/Physics2Behavior/physics2runtimebehavior.js

```diff
@@ -290,7 +307,55 @@ gdjs.Physics2RuntimeBehavior.prototype.createShape = function() {
     // Set the offset
     shape.set_m_p(this.b2Vec2(offsetX, offsetY));
   } else if (this.shape === 'Polygon') {
-    // Needs a vertices editor
+    shape = new Box2D.b2PolygonShape();
+    // Not convex, fall back to a box
+    if (!this.polygon.isConvex()) {
+      var width =
+        (this.owner.getWidth() > 0 ? this.owner.getWidth() : 1) *
+        this._sharedData.invScaleX;
+      var height =
+        (this.owner.getHeight() > 0 ? this.owner.getHeight() : 1) *
+        this._sharedData.invScaleY;
+      // Set the shape box
+      shape.SetAsBox(width / 2, height / 2, this.b2Vec2(offsetX, offsetY), 0);
+    } else {
+      var originOffsetX = 0;
+      var originOffsetY = 0;
+      if(this.polygonOrigin === "Origin"){
+        originOffsetX = (this.owner.getWidth() > 0 ? -this.owner.getWidth()/2 : 0)
+                        + (this.owner.getX() - this.owner.getDrawableX());
+        originOffsetY = (this.owner.getHeight() > 0 ? -this.owner.getHeight()/2 : 0)
+                        + (this.owner.getY() - this.owner.getDrawableY());
+      }
+      else if(this.polygonOrigin === "TopLeft"){
+        originOffsetX = this.owner.getWidth() > 0 ? -this.owner.getWidth()/2 : 0;
+        originOffsetY = this.owner.getHeight() > 0 ? -this.owner.getHeight()/2 : 0;
+      }
+      // Generate vertices if not done already
+      if (!this._verticesBuffer) {
+        // Store the vertices using a memory allocation function
+        var buffer = Box2D._malloc(
+          this.polygon.vertices.length * 8,
+          'float',
+          Box2D.ALLOC_STACK
+        );
+        this._verticesBuffer = buffer;
+      }
+      // Overwrite the vertices stored in the buffer
+      var offset = 0;
+      for (var i = 0, len = this.polygon.vertices.length; i < len; i++) {
+        Box2D.HEAPF32[(this._verticesBuffer + offset) >> 2] =
```

Comment:
Why the >> 2? I also found it on https://github.com/kripken/box2d.js/issues/87 but not sure exactly why it's useful when addressing the memory here.

Reply:
To divide by the size of float (4) and get the correct position, shift automatically deals with the decimal fraction.
It is a headache and would be a lot simpler if the support for `setValue` would not have been removed :(